File: frontend/src/services/notifications.js
const API_BASE_URL = 'http://localhost:8000/api';

export const notificationsService = {
    
    getNotifications: async (userId) => {
        try {
            const response = await fetch(`${API_BASE_URL}/notifications/?user_id=${userId}`);
            if (!response.ok) throw new Error('Failed to fetch notifications');

            return await response.json();
        } catch (error) {
            console.error('Error fetching notifications:', error);
            throw error;
        }
    },

    
    getUnreadCount: async (userId) => {
        try {
            const response = await fetch(`${API_BASE_URL}/notifications/unread-count?user_id=${userId}`);
            if (!response.ok) throw new Error('Failed to fetch unread count');

            return await response.json();
        } catch (error) {
            console.error('Error fetching unread count:', error);
            return { count: 0 };
        }
    },

    
    markAsRead: async (notificationId) => {
        try {
            const response = await fetch(`${API_BASE_URL}/notifications/${notificationId}/read`, {
                method: 'PUT',
            });

            if (!response.ok) throw new Error('Failed to mark notification as read');

            return await response.json();
        } catch (error) {
            console.error('Error marking notification as read:', error);
            throw error;
        }
    },
};

File: backend/app/routers/notifications.py
from typing import List
from fastapi import APIRouter, HTTPException, Query

from app.models.notification import Notification
from app.utils.database import (
    get_user_notifications,
    mark_notification_read,
    get_unread_count
)

router = APIRouter()

@router.get("/", response_model=List[Notification])
async def get_notifications(
    user_id: str = Query(..., description="ID of the user to fetch notifications for")
):

    return get_user_notifications(user_id)

@router.get("/unread-count")
async def get_unread_notification_count(
    user_id: str = Query(..., description="ID of the user")
):

    count = get_unread_count(user_id)
    return {"count": count}

@router.put("/{notification_id}/read", response_model=Notification)
async def read_notification(notification_id: str):

    notification = mark_notification_read(notification_id)
    
    if not notification:
        raise HTTPException(status_code=404, detail="Notification not found")
    
    return notification
